tests/views/test_survey.py

```diff
@@ -45,3 +48,20 @@ def test_survey_list_connection_error(self, mock_request):
 
         self.assertEqual(response.status_code, 200)
         self.assertIn("FAIL".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_survey_view(self, mock_request):
+        mock_request.get(url_get_survey_by_short_name, json=survey_info)
+
+        response = self.app.get("/surveys/bres")
+
+        self.assertEqual(response.status_code, 200)
+
+    @requests_mock.mock()
+    def test_survey_view_fail(self, mock_request):
+        mock_request.get(url_get_survey_by_short_name, status_code=500)
+
+        response = self.app.get("/surveys/bres")
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
Should the status_code be 200 when there is a failure?

Reply:
I'm not sure what best practice is here. This is where we'll be rendering error screens, if we successfully render the error screen do we return a failing status code with it?

Reply:
Yeah not massively sure. Can't find much around it. The main benefit I'd see is it will probably make testing easier when we test errors. Means we can just assert for http status codes vs parsing html?

Reply:
We'd likely still have to parse html as we'll have different error pages in the future (404, 500, more specific ones).